text/0000-ibm-liberty-buildpack.md

```diff
@@ -0,0 +1,68 @@
+# Paketo Community IBM Liberty Buildpack
+
+## Summary
+
+1. Create a Cloud Native Buildpack for the purpose of installing the IBM [OpenLiberty](https://openliberty.io/) Java runtime. 
+2. Create a new top-level multi buildpack that packages up Eclipse OpenJ9 & OpenLiberty for use by application developers.
+
+## Motivation
+
+A [ibm-websphere-liberty-buildpack](https://github.com/cloudfoundry/ibm-websphere-liberty-buildpack) exists in the cloudfoundry community and is widely used and popular.  This RFC will
+take that buildpack to the next level, cloud native.  This buildpack will eventually replace the v2 CF ibm-websphere-buildpack.   
+
+## Detailed Explanation
+
+### IBM OpenLiberty Buildpack
+
+This buildpack has the purpose of installing the IBM Open Liberty. It will will be based on the Paketo [apache-tomcat](https://github.com/paketo-buildpacks/apache-tomcat) buildpack. 
+
+This buildpack will participate if all the following conditions are met:
+
+- `<APPLICATION_ROOT>/WEB-INF` exists or
+-  A liberty server directory exists or
```

Comment:
Can we describe how a liberty server directory is identified? A server.xml file at the root of the app dir?

Reply:
The liberty server directory is identified when a server.xml exists in `--path`.    If you have an on-prem Liberty server already configured, `--path` would point to the server's configuration directory:  `$LIBERTY_HOME/wlp/usr/servers/serverName`.

For example, if the server name is `defaultServer`, the directory structure would look like this:

wlp/usr/servers/defaultServer/server.xml
wlp/usr/servers/defaultServer/app
wlp/usr/servers/defaultServer/app/MyApp.war

The buildpack would contribute the server.xml and war. 

https://cloud.ibm.com/docs/cloud-foundry-public?topic=cloud-foundry-public-options_for_pushing#server_directory

Reply:
>The buildpack would contribute the server.xml and war

I'm not sure I follow this point. Do you mean the user contributes? You're saying that a user would `pack build -p wlp/usr/servers/defaultServer`, so the user would provide server.xml & the WAR. Or am I missing something about what the buildpack would do?

Reply:
I'm also wondering, is the server.xml required? The first bullet here says 

>`<APPLICATION_ROOT>/WEB-INF` exists

which would mean that the user `pack build -p my/file.war` only and no server.xml. Is that something which would be supported?

Reply:
@dmikusa Yes, my intention is the user would `pack build -p wlp/usr/servers/defaultServer` and the buildpack would find the app and server.xml to contribute.  

The buildpack has a default server.xml that would be used if one is not provided.